app/src/main/java/com/example/levi9application/FavoritesAdapter.kt

```diff
@@ -0,0 +1,76 @@
+package com.example.levi9application
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.GridLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import com.example.levi9application.databinding.ItemFavoritesBinding
+import com.example.levi9application.databinding.ItemLabelBinding
+import com.example.levi9application.model.FavoriteRecyclerViewItem
+import com.example.levi9application.view.FavoritesHolder
+
+class FavoritesAdapter(private var items: MutableList<FavoriteRecyclerViewItem>) :
+    RecyclerView.Adapter<FavoritesHolder>() {
+
+    fun setItems(newItems: List<FavoriteRecyclerViewItem>){
+        items.apply{
+            clear()
+            addAll(newItems)
+        }
+        notifyDataSetChanged()
+    }
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): FavoritesHolder {
```

Comment:
We should not use R.layout.* for referencing viewTypes since this Int is not consistent and Android can change it overtime. Find different approach to handle this case. This comment applies for any place where we use viewType to handle the logic.